api/src/main/java/jakarta/data/metamodel/Attribute.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023,2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.Sort;
+
+/**
+ * Represents an entity attribute in the {@link StaticMetamodel}.
+ */
+public interface Attribute {
```

Comment:
I would merge this interface with `SortableAttribute`, I am assuming that the minimum of an attribute is being sortable.

Reply:
> Similarly, if an attribute is an embeddable or collection type, it might not make sense to be able to sort it either.

Ah yeah, of course. Forgot about that.

Reply:
> > I would merge this interface with `SortableAttribute`, I am assuming that the minimum of an attribute is being sortable.
> 
> We should not assume that every attribute is sortable. For example, the Jakarta Data specification lists `byte[]` as a data type that represents binary data. I would not expect that to be sortable. Similarly, if an attribute is an embeddable or collection type, it might not make sense to be able to sort it either. Having a interface to designate non-sortable types helps prevent the user during development time from introducing errors of requesting sort on things that cannot be sorted.

@njr-11 My point is that the database will define what is sortable or not, non the spec or the Java App.

Reply:
> @njr-11 My point is that the database will define what is sortable or not, non the spec or the Java App.

I agree. The database makes the decision. The Java App must be aware of that -- what is and is not sortable -- in order to write valid code asking to sort on some attributes and not others.  If the Java App tries to declare something sortable that cannot be sorted, that should result in an error.  Maybe we should write that somewhere in the spec.